File: recursion.py
def fact(x, y = 1):
    if x == 1:
        return 1 * y
    else:
        res = x * y
        return fact(x-1, res)
result = fact(6)
print(result)

total = sum([2, 4, 6])
print(total)

File: Recursion/recursion.js
function fact (x, y = 1) {
  if (x === 1) {
    return x * y;
  } else {
    let res = x * y;
    return fact(x - 1, res);
  }
}

console.log(fact(6))
console.log([2, 4, 6].reduce((pre, next, index, arr) => pre + next))
